### PR TITLE
Multinode xena

### PR DESCRIPTION
**Pull request by jackhodgkiss**

### CODE REVIEW

File: .automation.conf/config.sh [hunk 1]
```diff
@@ -28,3 +28,6 @@ fi
 if [[ -f ${KAYOBE_AUTOMATION_REPO_ROOT}/etc/kolla/public-openrc.sh ]]; then
     export TEMPEST_OPENRC="$(< ${KAYOBE_AUTOMATION_REPO_ROOT}/etc/kolla/public-openrc.sh)"
 fi
+
+export KAYOBE_AUTOMATION_TEMPEST_LOADLIST=tempest-full
+export KAYOBE_AUTOMATION_TEMPEST_SKIPLIST=tempest-full
```
Comment (markgoddard) [inline .automation.conf/config.sh]: Let's keep this file as it was.

File: .automation.conf/tempest/tempest-ci-multinode.overrides.conf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+[compute-feature-enabled]
+resize = true
+live_migration = true
+block_migration_for_live_migration = false
+volume_backed_live_migration = true
+console_output = true
+
+[volume]
+min_microversion = 3.0
+max_microversion = 3.64
```
Comment (markgoddard) [inline .automation.conf/tempest/tempest-ci-multinode.overrides.conf]: 3.66 for xena

File: etc/kayobe/ansible/requirements.yml [hunk 3]
```diff
@@ -1,8 +1,10 @@
 ---
```
Comment (markgoddard) [inline etc/kayobe/ansible/requirements.yml:1]: No need to change this file.

File: etc/kayobe/environments/ci-multinode/README.md [hunk 4]
```diff
@@ -1,48 +1,233 @@
 # Multinode Test Environment
```
Comment (markgoddard) [inline etc/kayobe/environments/ci-multinode/README.md]: I moved this file to the new documentation, doc/source/contributor/environments/ci-multinode.rst.

File: etc/kayobe/environments/ci-multinode/kolla/config/keepalived/keepalived.conf [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+vrrp_instance kolla_internal_vip_{{ keepalived_virtual_router_id }} {
+    state BACKUP
+    nopreempt
+    interface {{ api_interface }}
+    virtual_router_id {{ keepalived_virtual_router_id }}
+    priority {{ groups['haproxy'].index(inventory_hostname) + 1 }}
+    advert_int 3 # Increase advert interval from 1 to 3 as there was a lot of jumping amongst controllers on the internal network 
+{% if keepalived_traffic_mode == 'unicast' %}
+    unicast_src_ip {{ api_interface_address }}
+{% if groups['haproxy'] | length > 1 %}
```
Comment (bbezak) [inline etc/kayobe/environments/ci-multinode/kolla/config/keepalived/keepalived.conf]: haproxy group was renamed to load balancer

File: etc/kayobe/environments/ci-multinode/kolla/config/keepalived/keepalived.conf [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    unicast_peer {
+{% for host in groups['haproxy'] %}
```
Comment (bbezak) [inline etc/kayobe/environments/ci-multinode/kolla/config/keepalived/keepalived.conf]: same here

**Review by bbezak — CHANGES_REQUESTED** [review]

File: .automation.conf/config.sh [hunk 7]
```diff
@@ -3,7 +3,7 @@
 
 # See: https://github.com/stackhpc/docker-rally/blob/master/bin/rally-verify-wrapper.sh for a full list of tempest parameters that can be overriden.
 # You can override tempest parameters like so:
-export TEMPEST_CONCURRENCY=8
+export TEMPEST_CONCURRENCY=6
```
Comment (markgoddard) [inline .automation.conf/config.sh]: Need to bear in mind this is a shared file that affects all environments.

File: etc/kayobe/kolla.yml [hunk 8]
```diff
@@ -457,7 +457,7 @@ kolla_build_args:
 #kolla_enable_ceilometer_horizon_policy_file:
 #kolla_enable_ceilometer_ipmi:
 #kolla_enable_cells:
-kolla_enable_central_logging: true
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: Please don't modify these global flags.

File: .automation.conf/tempest/load-lists/tempest-full [hunk 9]
```diff
@@ -0,0 +1,1601 @@
+tempest.api.object_storage.test_object_slo.ObjectSloTest.test_upload_manifest[id-2c3f24a6-36e8-4711-9aa2-800ee1fc7b5b]
```
Comment (markgoddard) [inline .automation.conf/tempest/load-lists/tempest-full:1]: I do wonder how portable this list is between different environments. Probably something to ask about in Slack. i.e. do different environments tend to have different "full" test lists, and if so, should they?

File: .automation.conf/tempest/skip-lists/tempest-full [hunk 10]
```diff
@@ -0,0 +1 @@
+tempest.scenario.test_network_basic_ops.TestNetworkBasicOps.test_subnet_details.*: "Cirros image doesn't have '/var/run/udhcpc.eth0.pid"
```
Comment (markgoddard) [inline .automation.conf/tempest/skip-lists/tempest-full]: The skip list is probably fairly environment-specific. We might need to give it a multinode-specific name, and conditionally use it in config.sh.

File: etc/kayobe/ansible/requirements.yml [hunk 11]
```diff
@@ -1,7 +1,7 @@
...
 collections:
   - name: stackhpc.cephadm
-    version: 1.10.0
+    version: 1.12.1
```
Comment (markgoddard) [inline etc/kayobe/ansible/requirements.yml]: Is this required for a bug fix or something? Probably deserves a quick release note.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: .automation.conf/config.sh (hunk 7 above)
Comment (jackhodgkiss) [reply .automation.conf/config.sh]: Could we not use conditional statements to test the `KAYOBE_ENVIRONMENT`? 

if `ci-multinode` 6 else 8.

File: etc/kayobe/ansible/requirements.yml (hunk 11 above)
Comment (jackhodgkiss) [reply etc/kayobe/ansible/requirements.yml]: Added small release note to reflect the bump in version.

File: etc/kayobe/kolla.yml (hunk 8 above)
Comment (jackhodgkiss) [reply etc/kayobe/kolla.yml]: This has been reverted and the disabling of services has been moved into `ci-multinode/kolla.yml`

**Comment by markgoddard** [comment]
pep8 is failing

File: etc/kayobe/ansible/requirements.yml (hunk 11 above)
Comment (markgoddard) [inline etc/kayobe/ansible/requirements.yml]: I'm still concerned about this. It introduces the `cephadm_host_labels` bug which is going to affect all deployments and is annoying to diagnose. I'd much rather we had a fix in the collection.

File: .automation.conf/config.sh (hunk 7 above)
Comment (markgoddard) [reply .automation.conf/config.sh]: Looks like the original value was 2

File: .automation.conf/tempest/skip-lists/tempest-full (hunk 10 above)
Comment (markgoddard) [reply .automation.conf/tempest/skip-lists/tempest-full]: Still a problem

File: .automation.conf/tempest/load-lists/tempest-full (hunk 9 above)
Comment (markgoddard) [reply .automation.conf/tempest/load-lists/tempest-full:1]: I've asked about this in slack

File: etc/kayobe/ansible/requirements.yml (hunk 11 above)
Comment (markgoddard) [reply etc/kayobe/ansible/requirements.yml]: Probably the way to fix it is using `| default([])` when referencing the variable in the collection.

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Comment by jackhodgkiss** [comment]
> pep8 is failing

Resolved this issue ran locally before pushing.

File: .automation.conf/config.sh [hunk 12]
```diff
@@ -19,6 +19,11 @@ if [ ! -z ${KAYOBE_ENVIRONMENT:+x} ]; then
     # Seem to get servers failing to spawn with higher concurrency
     export TEMPEST_CONCURRENCY=1
   fi
+  if [[ "$KAYOBE_ENVIRONMENT" =~ "ci-multinode" ]]; then
+      # Increase concurrency for multinode environment. Eight has proven to high, leading to failing tests due to poor CEPH performance.
```
Comment (markgoddard) [inline .automation.conf/config.sh]: Comment no longer relevant

File: releasenotes/notes/cephadm-bump-294148e93121da48.yaml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    Increase `stackhpc.cephadm` collection to version `1.12.1`.
```
Comment (markgoddard) [inline releasenotes/notes/cephadm-bump-294148e93121da48.yaml]: 1.12.2

File: releasenotes/notes/bump-cephadm-version-e22b37907acf0efd.yaml [hunk 14]
```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Increase the required version of `stackhpc.cephadm` collection to avoid an
+    issue where deployment of cephadm would fail if `cephadm_host_labels` was 
+    absent from group or host vars.
```
Comment (markgoddard) [inline releasenotes/notes/bump-cephadm-version-e22b37907acf0efd.yaml]: This bug wasn't in the 1.10.0 release, so no need for a reno

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: releasenotes/notes/bump-cephadm-version-e22b37907acf0efd.yaml (hunk 14 above)
Comment (jackhodgkiss) [reply releasenotes/notes/bump-cephadm-version-e22b37907acf0efd.yaml]: Removed.

File: releasenotes/notes/cephadm-bump-294148e93121da48.yaml (hunk 13 above)
Comment (jackhodgkiss) [reply releasenotes/notes/cephadm-bump-294148e93121da48.yaml]: Fixed.

File: .automation.conf/config.sh (hunk 12 above)
Comment (jackhodgkiss) [reply .automation.conf/config.sh]: Removed irrelevant comment.

**Review by markgoddard — APPROVED** [review]
Good work Jack.

**Pull request merged**